code/modules/projectiles/ammo_datums.dm

```diff
@@ -332,7 +332,13 @@ GLOBAL_LIST_INIT(no_sticky_resin, typecacheof(list(/obj/item/clothing/mask/faceh
 	sundering = 3
 
 /datum/ammo/bullet/revolver/on_hit_mob(mob/M,obj/projectile/P)
-	staggerstun(M, P, stagger = 1, slowdown = 0.5, knockback = 1, shake = 0.5)
+	var/obj/item/weapon/gun/shot_from = P.shot_from
+	var/stagger_knock = 0 // snub the snubnose stagger
+	if(!shot_from.attachable_allowed?.Find(/obj/item/attachable/standard_revolver_longbarrel))
+		stagger_knock = 1 // revolvers without the attachment always can :/
+	if(shot_from.attachments?.Find(ATTACHMENT_BARREL_MOD))
```

Comment:
this is yucky and needs a better solution

Ideally use signals or snubnosed would only change a var without having to go backwards and check if there is a barrel on the revolver

Reply:
where do I put the signals?

Reply:
This is done